promql/test.go

```diff
@@ -86,7 +104,7 @@ func newTestFromFile(t testutil.T, filename string) (*Test, error) {
 }
 
 // QueryEngine returns the test's query engine.
-func (t *Test) QueryEngine() *Engine {
+func (t *Test) QueryEngine() TestEngine {
```

Comment:
If we want this, I would suggest having `NewTest` expecting user to pass engine to it and use the passed instance (not assume Test will give it back), so

* QueryEngine method is removed
* NewEngine(EngineQuerier) method only

This is will simplify things overall I think.

Reply:
That does sound appealing, what i dont get though is:`Test` has a `clear()` method that constructs a new engine currently using `NewEngine`; we cannot model this if we pass en engine externally right?

Reply:
I wonder, clear just seems to reset options on the engine; resetting the storage seems to be the important part; ill try your suggestion!

Reply:
👍🏽

Reply:
I noticed that most usages of the Test are only used to execute the Load command. I took out the engine of the test and added it to `Run` as parameter if the test manifest uses an engine to execute an assertion.